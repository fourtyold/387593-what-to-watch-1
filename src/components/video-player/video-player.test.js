import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player.jsx";

const options = {
  film: {
    name: `Moonrise Kingdom`,
    page: `https://moonrise.html`,
    genre: `Adventure`,
    previewImage: `https://preview_1.jpg`,
    previewVideoLink: `https://preview_1.mp4`
  },
  isPlaying: false,
  muted: true
};

function createNodeMock(element) {
  if (element.type === `video`) {
    return {};
  }
  return null;
}

it(`Video player correctly renders`, () => {
  const customMockCreator = {createNodeMock};
  const tree = renderer.create(
      <VideoPlayer
        film={options.film}
        isPlaying={options.isPlaying}
        muted={options.muted}
      />, customMockCreator).toJSON();

  expect(tree).toMatchSnapshot();
});
