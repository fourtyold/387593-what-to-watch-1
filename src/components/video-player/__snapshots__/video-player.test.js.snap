// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video player correctly renders 1`] = `
<div>
  <video />
</div>
`;
